File: pkg/dart2cpp/sample/cpp_generated/test_return_this.cpp
#include "dart2cpp.h"

// 工具宏定义

// ============================================================================
// 类: Builder
// ============================================================================

class Builder {
private:
  String _value = dart_string("");
public:
  Builder() {
  }
  
  ObjectPtr<Builder> add(String text) {
    this->_value = (this->_value + text);
return ObjectPtr<std::remove_reference_t<decltype(*this)>>(this);
  }
  
  String build() {
    return this->_value;
  }
  
};

// ============================================================================
// 主函数
// ============================================================================

int main() {
  try {
    auto builder = ObjectPtr<Builder>(new Builder())->add(dart_string("Hello"))->add(dart_string(" "))->add(dart_string("World"));
dart_print(builder->build());
    return 0;
  } catch (const std::exception& e) {
    std::cerr << "Error: " << e.what() << std::endl;
    return 1;
  }
}

File: pkg/dart2cpp/sample/dart/test_return_this.dart
class Builder {
  String _value = '';
  
  Builder add(String text) {
    _value += text;
    return this;
  }
  
  String build() => _value;
}

void main() {
  var builder = Builder()
    .add('Hello')
    .add(' ')
    .add('World');
  print(builder.build());
}